tracer/test/Datadog.Trace.TestHelpers/SettingsTestsBase.cs

```diff
@@ -0,0 +1,86 @@
+// <copyright file="SettingsTestsBase.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.TestHelpers
+{
+    public abstract class SettingsTestsBase
+    {
+        public enum Strings
+        {
+            /// <summary>
+            /// Empty string values are accepted
+            /// </summary>
+            AllowEmpty,
+
+            /// <summary>
+            /// Empty string values are replaced by the default value
+            /// </summary>
+            DisallowEmpty
+        }
+
+        public static TheoryData<string, bool?> BooleanTestCases(bool? defaultValue) 
+            => new TheoryData<string, bool?>
+            {
+                { "true", true },
```

Comment:
in ToBoolean() in StringExtensions, we are also allowing `t`, `y` as true and `f`, `n` as false, but not sure if it can be included in those tests cases cause I'm not sure ToBoolean is used everywhere. Just mentioning as I haven't fully checked